tree/account_tree.go

```diff
@@ -255,16 +259,19 @@ func AccountToNode(
 	return hashVal, nil
 }
 
-func NewEmptyAccountAssetTree(
+func NewEmptyAccountAssetTreeFunc(
 	ctx *Context,
 	index int64,
 	blockHeight uint64,
-) (tree bsmt.SparseMerkleTree, err error) {
-	return bsmt.NewBASSparseMerkleTree(
-		bsmt.NewHasher(mimc.NewMiMC()),
-		SetNamespace(ctx, accountAssetNamespace(index)),
-		AssetTreeHeight, NilAccountAssetNodeHash,
-		ctx.Options(int64(blockHeight))...)
+) func() bsmt.SparseMerkleTree {
+	return func() bsmt.SparseMerkleTree {
+		tree, _ := bsmt.NewBASSparseMerkleTree(
```

Comment:
It is possible to return error from `NewBASSparseMerkleTree`, please handle it properly, or throw the error to caller.